bundles/io/org.eclipse.smarthome.io.net/src/main/java/org/eclipse/smarthome/io/net/http/HttpRequestBuilder.java

```diff
@@ -0,0 +1,127 @@
+/**
+ * Copyright (c) 2014,2018 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.eclipse.smarthome.io.net.http;
+
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.charset.StandardCharsets;
+import java.util.Properties;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Builder class to construct http requests
+ *
+ * @author Martin van Wingerden - initial API
+ */
+public class HttpRequestBuilder {
+    private static final int DEFAULT_TIMEOUT_IN_MS = 5_000;
+
+    private final String method;
+    private final String url;
+    private InputStream inputStream;
+    private int timeoutMilliseconds = DEFAULT_TIMEOUT_IN_MS;
+    private Properties headers;
+    private String contentType;
+
+    /**
+     * Private constructor, to hide away the construction behind some factory methods
+     *
+     * @param method Http method to be used
+     * @param url url to fetch or post data to
+     */
+    private HttpRequestBuilder(String method, String url) {
+        this.method = method;
+        this.url = url;
+    }
+
+    /**
+     * Construct an http request builder to get data from an url
+     *
+     * @param url to fetch the data from
+     * @return a request builder to construct and complete the request
+     */
+    public static HttpRequestBuilder getFrom(String url) {
+        return new HttpRequestBuilder(HttpUtil.METHOD_GET, url);
+    }
+
+    /**
+     * Construct an http request builder to post data to a url
+     *
+     * @param url to post data to
+     * @return a request builder to construct and complete the request
+     */
+    public static HttpRequestBuilder postTo(String url) {
+        return new HttpRequestBuilder(HttpUtil.METHOD_POST, url);
+    }
+
+    /**
+     * Add content to this request
+     *
+     * @param content a string containing the data to be pushed to the url
+     * @return a request builder to construct and complete the request
+     */
+    public HttpRequestBuilder withContent(String content) {
+        this.inputStream = new ByteArrayInputStream(content.getBytes(StandardCharsets.UTF_8));
+        return this;
+    }
+
+    /**
+     * Add an additional header to the request
+     *
+     * @param header name of the header, eg. Content-Type
+     * @param value value of the header, eg. "application/json"
+     * @return a request builder to construct and complete the request
+     */
+    public HttpRequestBuilder withHeader(String header, String value) {
```

Comment:
The invariants (`Objects#requireNonNull(..)` or JDT Annotations)

Reply:
Done